src/wombats/handlers/user.clj

```diff
@@ -196,18 +208,33 @@
    (fn [{:keys [request response] :as context}]
      (let [update-wombat (dao/get-fn :update-user-wombat context)
            get-wombat (dao/get-fn :get-wombat-by-id context)
+           get-entire-user-by-id (dao/get-fn :get-entire-user-by-id context)
            {wombat-id :wombat-id
             user-id :user-id} (:path-params request)
            wombat (merge (:edn-params request)
-                         {:wombat/id wombat-id})]
+                         {:wombat/id wombat-id})
+           user (get-entire-user-by-id user-id)]
 
        (sutils/validate-input ::wombat-params wombat)
 
+       (when-not user
+         (handler-error (str "User '" user-id "' does not exist.")))
+
        (when-not (user-owns-wombat? user-id wombat-id context)
          (authorization-error "Cannot update this wombat"))
 
-       @(update-wombat wombat)
-
-       (assoc context :response (assoc response
-                                       :status 200
-                                       :body (get-wombat wombat-id)))))))
+       (let [url (str github-repo-api-base (:wombat/url wombat))
+             auth-headers {:headers {"Accept" "application/json"
+                                     "Authorization" (str "token "
+                                                          (:user/github-access-token user))}}
+             {status :status} @(http/get url auth-headers)]
+
+         (if (= status 200)
+           (do
+             @(update-wombat wombat)
+             (assoc context :response (assoc response
+                                             :status 200
+                                             :body (get-wombat wombat-id))))
+           (handler-error
+            (str "Oh no!!! " (:wombat/name wombat) " is homeless! "
+                 "Check you url '" (:wombat/url wombat) "' and make sure you have code at that location."))))))))
```

Comment:
Might be good to pull out error messages as well so we can easily change them if we want (fbf to the cms)